### ISSUE TITLE
Research which AI model to use

### ISSUE THREAD

**Issue by lupinteq** [GIT-ATHON-GROUP/AI-CareerGuide-DocumentationRepo#1]
Conduct research to determine which AI MODEL will be best suited to our project and explain how we will use the model to build our application.

**Comment by Fhulu1** [comment]
I think we should as use expert systems as it also assists with decision-making abilities.

**Comment by Tshepo-Mzimba** [comment]
Guys the currently we have four main types of AI, which are REACTIVE MACHINES, LIMITED MEMORY, and other two that does not exist yet.
- Reactive Machines are AI systems that have no memory and are task specific, meaning that an input always delivers the same output
- Limited memory imitates the way our brains' neurons work together, meaning it gets smarter as it receive more data to train on.
So which one are we going to consider

**Comment by CodeboyTF** [comment]
with what you just mentioned i feel like the LIMITED MEMORY sounds more advantageous than the other because it studies past data to improve decision making from its memory
now if im getting this correctly we looking for/or creating an AI system which will help with the case study we chose right?which is the education industry(helping high scholars to chose their desired tertiary course to help develop their goals.)
which by limited memory it can learn from past data like how(course requuirements entry levels(Aps) and such,course aims and biography of the course)
for example if a high scholar wants to study (informatio technology)our AI system will or have to recomend the course requirement in order to be admitted and also give insights and the necessary information and guidance of the course to the high scholar.
[if im not on the right track can i be helped understand]

**Comment by NathanTshiteya** [comment]
Limited memory machines are machines that are capable of learning from historical data to make decisions. Nearly all existing applications that we know of come under this category of AI. All present-day AI systems, such as those using deep learning, are trained by large volumes of training data that they store in their memory to form a reference model for solving future problems. For instance, an image recognition AI is trained using thousands of pictures and their labels to teach it to name objects it scans. When an image is scanned by such an AI, it uses the training images as references to understand the contents of the image presented to it, and based on its “learning experience” it labels new images with increasing accuracy.

Almost all present-day AI applications, from chatbots and virtual assistants to self-driving vehicles are all driven by limited memory AI.

That is what I found by making some research about Limited memory, and I think it's interesting.

**Comment by MashauLW** [comment]
here are some of advantages of limited memory
1.Optimized Performance: Efficient memory management can lead to better performance of your software. It ensures that memory is allocated and deallocated appropriately, reducing the chances of memory leaks or excessive memory usage that could slow down your application.

2.Stability: Proper memory management helps prevent crashes and unexpected behavior caused by memory-related issues, such as accessing uninitialized memory or overwriting memory outside the intended bounds.

3.Resource Efficiency: By using memory judiciously, you can make your application more resource-efficient. This is especially important for software running on resource-constrained devices or in cloud environments where minimizing memory usage can reduce costs.

so i think it can be suitable for our project

**Comment by MashauLW** [comment]
i think we must use machine learning algorithms as our AI model
and here is how we gonna use it to build our application 
1. Define the problem you're trying to solve and the goals of your application.
2. Gather the data you need to train your model, and clean and preprocess it.
3. Choose a machine learning algorithm that's suitable for your problem (e.g. linear regression, decision trees, neural networks).
4. Train your model on your training data, and evaluate its performance on test data.
5. Tweak and optimize your model until you're happy with its performance.
i stand to be corrected

**Comment by OctoberTwenty4th** [comment]
Good day guys, I also think the machine learning algorithms are a great go, it'll also come at an advantage when we want to try out models such as speech recognition on our project, I'm not sure if it's necessary to include it, other than that it will help a lot with complex classification of data, and the fact that it can trace back previous data and monitor new data overtime makes it a great tool.... i like the fact that over time, the machine builds itself from what is has learned.

**Comment by lupinteq** [comment]
According to my findings, the AI Career guide software will require multiple components to function well.

The AI Career Guide is an Interpersonal intelligence software that recommends careers based on user interests.

1. The AI Career Guide will best work using Unsupervised machine learning algorithms, meaning the AI software will not have any supervisor to provide guidance.

2. Use K-Means clustering to group careers based on user interests to provide an unbiased perspective on career choices .

3. Use the K-Nearest Neighbours (KNN) algorithm to find similar career choices.

4. Use Neural Networks and K-Means clustering for Unsupervised learning and for recommending careers

**Comment by Fhulu1** [comment]
I think we should combine machine learning and expert systems 
Expert systems is a type of AI that can match decision-making abilities as a human which is would work well with our system. Because our system is for career guidance it expert systems would kind of match the decision-making abilities of a career guidance counsellor.

**Comment by Fhulu1** [comment]
I see you guys said that limited memory stores pasted data to improve decision-making, what pasted data will our system store so that it can be able to make decisions and help students with their career choices.

**Comment by lupinteq** [comment]
Solved!
